gui/markdown-viewer.js

```diff
@@ -0,0 +1,76 @@
+const React = require('react')
+const ReactDOM = require('react-dom')
+const ReactMarkdown = require('react-markdown')
+const e = React.createElement
+
+class Banner extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      closed: false
+    }
+
+    this.close = this.close.bind(this)
+  }
+
+  close() {
+    this.setState({ closed: true })
+  }
+
+  render() {
+    const { translate, level, title, details } = this.props
```

Comment:
Since you're used to have type checking with Flow, you can use something like https://www.npmjs.com/package/prop-types to check and warn about the type you're using. 

It's far from flow or TS, but at least, you can define the "API" of the component. 

Banner.propTypes = {
  translate: PropTypes.func.required, 
  level: PropTypes...
}

Reply:
Ah yes I could use PropTypes!